### PR TITLE
Updated an example custom "actions" of component List

### PR DESCRIPTION
**Pull request by Artexoid**
The previous version of example in documentation was not actual and didn't work in version 3+. I spent an hour to understand it, make a research and get the code from sources. I hope it will help community.

### CODE REVIEW

File: docs/List.md [hunk 1]
```diff
@@ -86,46 +86,86 @@ The title can be either a string, or an element of your own.
...
-            filter={filterValues}
-            exporter={exporter}
-        />
-        {/* Add your custom actions */}
-        <Button color="primary" onClick={customAction}>Custom Action</Button>
-    </Toolbar>
+    maxResults,
+    total,
+    ...rest
+}) => useMemo(
```
Comment (fzaninotto) [inline docs/List.md]: please remove the `useMemo` to make the code more readable

File: docs/List.md [hunk 2]
```diff
@@ -86,46 +86,86 @@ The title can be either a string, or an element of your own.
...
+                label="Show calendar"
+            >
+                <IconEvent />
+            </Button>
+        </TopToolbar>
+    ),
+    [resource, displayedFilters, filterValues, selectedIds, filters, total],
 );
 
+ListActions.propTypes = {
```
Comment (fzaninotto) [inline docs/List.md]: please remove the propTypes to make the code more readable

File: docs/List.md [hunk 3]
```diff
@@ -86,46 +86,86 @@ The title can be either a string, or an element of your own.
...
+    () => (
+        <TopToolbar className={className} {...sanitizeListRestProps(rest)}>
+            {filters && cloneElement(filters, {
+                resource,
+                showFilter,
+                displayedFilters,
+                filterValues,
+                context: 'button',
+            })}
+            {hasCreate && <CreateButton basePath={basePath} />}
```
Comment (fzaninotto) [inline docs/List.md]: please simplify the example by removing the test on `hasCreate` (as in the current version of the code).

File: docs/List.md [hunk 4]
```diff
@@ -86,46 +86,86 @@ The title can be either a string, or an element of your own.
...
+            {exporter !== false && (
```
Comment (fzaninotto) [inline docs/List.md]: same here, please remove the test

**Comment by Artexoid** [comment]
@fzaninotto updated.

**Comment by fzaninotto** [comment]
Thanks!

**Comment by liviozanol** [comment]
You just saved my soul! Thank you very much!

**Pull request merged**